docs/reference/module-overview.md

```diff
@@ -0,0 +1,284 @@
+
+```javascript
+// modules/article/index.js
+modules.export = {
+  fields: {
+    add: {
+      // ...
+    },
+    group: {
+      meta: { // 👈 The group's identifying name is the object key.
+        label: 'Article metadata',
+        fields: [
+          'subtitle',
+          'author',
+          '_category'
+        ]
+      }
+    }
+  }
+};
+```
+
+### `filters`
+
+In piece type modules, the `filters` setting configures the pieces manager modal, adding and removing filtering interface (to view only certain pieces). `trash` and `visibility` filters are included by default.
+
+The `filters` object is configured with subsections: `add` and `remove`. Filters must correspond to an existing fields name or custom [query builder](#queries-self) on the piece type.
+
+#### `add`
+
+An object of filters to add to the piece type. Each filter is an object with its own configuration. Filter properties include:
+
+| Property | Description |
+| ------- | ------- |
+| `label` | Recommended, but not required, to customize the human-readable label. |
+| `inputType` | Choose an input field type for the filter from the following: `radio`, `checkbox`, or `select`. Defaults to `select`. |
+| `choices` | Manually set an array of choices instead of letting Apostrophe find the valid choices from the database. Choices include `label` and `value` properties. |
+| `def` | The default value for the manager filter. |
+
+```javascript
+// modules/article/index.js
+modules.export = {
+  filters: {
+    add: {
+      _category: { // 👈 Referencing a relationship field named `_category`
+        label: 'Article category'
+      },
+      featured: { // 👈 Referencing a boolean field name `featured`
+        labeled: 'Featured',
+        inputType: 'checkbox',
+        def: true,
+        choices: [
+          { value: true, label: 'Show featured' },
+          { value: false, label: 'Hide featured' }
+        ]
+      }
+    }
+  }
+};
+```
+
+#### `remove`
+
+An array of filter names from the base class module to remove.
+
+```javascript
+// modules/spotlight-article/index.js
+modules.export = {
+  extend: 'article',
+  filters: {
+    remove: [ 'featured' ]
+  }
+};
+```
+
+### `columns`
+
+In piece type modules, the `columns` setting configures the pieces manager modal, adding and removing the piece listing columns. Default columns include `title`, `updatedAt`, and `visibility`.
+
+#### `add`
+
+An object of columns to add to the piece type manager. Each column is an object with its own configuration. Column properties include:
+
+| Property | Description |
+| ------- | ------- |
+| `label` | Recommended, but not required, to show a column header label. |
+| `component` | An advanced option to use a custom Vue component for table cells in this column. See core components `AposCellBasic` (default) and `AposCellDate` for examples. |
```

Comment:
flagging that we should link to this when we can. Is there a master list of these options somewhere? Am I thinking about this wrong?